deepchem/molnet/load_function/material_datasets.py

```diff
@@ -0,0 +1,359 @@
+        }
+    },
+    **kwargs) -> Tuple[List, Tuple, List]:
+  """Load perovskite dataset.
+
+  Contains 18928 perovskite structures and their formation energies.
+  
+  Parameters
+  ----------
+  featurizer : StructureGraphFeaturizer
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  transformers : List{List of allowed transformers for this dataset}
+    A transformer that inherits from deepchem.trans.Transformer.
+  splitter : RandomSplitter
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload : bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  data_dir : str, optional
+    Path to datasets.
+  save_dir : str, optional
+    Path to featurized datasets.
+  featurizer_kwargs : dict
+    Specify parameters to featurizer, e.g. {"size": 1024}
+  splitter_kwargs : dict
+    Specify parameters to splitter, e.g. {"seed": 42}
+  transformer_kwargs : dict
+    Maps transformer names to constructor arguments, e.g.
+    {"BalancingTransformer": {"transform_x":True, "transform_y":False}}
+  **kwargs : additional optional arguments.
+
+  Returns
+  -------
+  tasks, datasets, transformers : tuple
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  References
+  ----------
+  .. [1] Castelli, I. et al. "New cubic perovskites for one- and two-photon water splitting using the computational materials repository." Energy Environ. Sci., (2012), 5, 9034-9043 DOI: 10.1039/C2EE22341D.
+
+  .. [2] Dunn, A. et al. "Benchmarking Materials Property Prediction Methods: The Matbench Test Set and Automatminer Reference Algorithm." https://arxiv.org/abs/2005.00707 (2020)
+
+  Examples
+  --------
+  >> import deepchem as dc
+  >> tasks, datasets, transformers = dc.molnet.load_perovskite(reload=False)
+  >> train_dataset, val_dataset, test_dataset = datasets
+  >> n_tasks = len(tasks)
+  >> n_features = train_dataset.get_data_shape()[0]
+  >> model = dc.models.MultitaskRegressor(n_tasks, n_features)
+
+  """
+
+  # Featurize
+  logger.info("About to featurize perovskite dataset.")
+  my_tasks = ['e_form']  # machine learning targets
+
+  # Get DeepChem data directory if needed
+  if data_dir is None:
+    data_dir = DEFAULT_DIR
+  if save_dir is None:
+    save_dir = DEFAULT_DIR
+
+  # Check for str args to featurizer and splitter
+  if isinstance(featurizer, str):
+    featurizer = DEFAULT_FEATURIZERS[featurizer](**featurizer_kwargs)
+  elif issubclass(featurizer, Featurizer):
+    featurizer = featurizer(**featurizer_kwargs)
+
+  if isinstance(splitter, str):
+    splitter = DEFAULT_SPLITTERS[splitter]()
+  elif issubclass(splitter, Splitter):
+    splitter = splitter()
```

Comment:
Same here

Reply:
Changed to check for `Splitter` subclass.